### PR TITLE
Page types deleting

### PR DESCRIPTION
**Pull request by orzechdev**
I want to merge this change because... it adds:
- page type deleting on page types details page
- bulk page types deleting on page types list page

Note: it contains changes introduced in https://github.com/mirumee/saleor-dashboard/pull/808.

**PR intended to be tested with API branch:** add-has-pages-field-to-page-type
https://github.com/mirumee/saleor/pull/6375

### Screenshots

In list view:

Zrzut ekranu 2020-11-2 o 16 28 43
Zrzut ekranu 2020-11-2 o 16 28 54
Zrzut ekranu 2020-11-2 o 16 29 09
Zrzut ekranu 2020-11-2 o 16 29 19

In details view:
Zrzut ekranu 2020-11-2 o 16 29 45
Zrzut ekranu 2020-11-2 o 16 31 35

### Pull Request Checklist

1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Translated strings are extracted.
1. [ ] Number of API calls is optimized.
1. [ ] The changes are tested.
1. [ ] Data-test are added for new elements.
1. [ ] Type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1175 Handle bulk deleting of page types on the page type list view](https://app.clickup.com/t/2549495/SALEOR-1175)

**Comment by patrys** [comment]
Task linked: [SALEOR-1184 Handle deleting page types on the page type details view](https://app.clickup.com/t/2549495/SALEOR-1184)

File: src/pageTypes/views/PageTypeList/PageTypeList.tsx [hunk 1]
```diff
@@ -118,6 +125,41 @@ export const PageTypeList: React.FC<PageTypeListProps> = ({ params }) => {
...
+  const hanldePageTypeBulkDelete = () =>
+    pageTypeBulkDelete({
+      variables: {
+        ids: params.ids
+      }
+    });
+
+  const selectedPageTypesHasPages = data?.pageTypes.edges.some(
+    pageType =>
+      pageType.node.hasPages && params.ids?.some(id => id === pageType.node.id)
```
Comment (dominik-zeglen) [inline src/pageTypes/views/PageTypeList/PageTypeList.tsx:160]: Is second check necessary?

**Review by dominik-zeglen — APPROVED** [review]

File: src/pageTypes/views/PageTypeList/PageTypeList.tsx (hunk 1 above)
Comment (orzechdev) [reply src/pageTypes/views/PageTypeList/PageTypeList.tsx:160]: Yes because we want to check only those page types which are selected.

**Pull request merged**